Implemented a minimal init command using the templates

etorreborre opened the pull request:
This is very basic, just a bridge between `./cabal` and the templates on Github.

domdere commented:
hmmm, this wont work as open source i think....

olorin commented:
Could this command be added to the usage text as well?

domdere commented:
(our templates are private atm)

etorreborre commented:
@domdere we are already referencing some private things in this file (`mafia`). I think that "open-source" doesn't always mean "fully executable". People can also fork and modify, or we can open source our templates in the future.

@olorin `init` was already in the usage section, but not wired to anything! In the fixup I add a bit more doc on what `init` does.

domdere commented:
> "open-source" doesn't always mean "fully executable"

thats disappointing :(.

would it be so terrible to have the init comment take in `ambiata/haskell-foo.template" instead of "foo"?

nhibberd commented:
Not actually reference mafia yet, just using it a the cache for alex and happy executables. Name is just for forwards compatibility (future open source project etc)

markhibberd commented on src/cabal:
```diff
@@ -437,6 +437,17 @@ run_update () {
+
+    mkdir tmp
+    git clone http://github.com/ambiata/haskell-$1.template tmp
+    cp -r tmp/* .
```
Think you will be losing dotfiles here.

etorreborre commented:
@domdere I changed the args so that `init` now takes the repository path like `ambiata/haskell-library.template`

@markhibberd I fixed the copying of dot files

domdere commented:
> I changed the args so that init now takes the repository path like ambiata/haskell-library.template

Thanks, sorry for whinging

:palm_tree: :shell: :sunglasses: :ocean:
